### PR TITLE
Update model-card for Autofomer

### PR DESCRIPTION
**Pull request by AnupKulkarniVK**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/model_doc/autoformer.md [hunk 1]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
 # Autoformer
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[Autoformer](https://huggingface.co/papers/2106.13008) is a time series forecasting model that rethinks Transformers with a twist — instead of just using attention, it brings in a **decomposition-based architecture** and a new **Auto-Correlation mechanism** to capture complex patterns in time series data.
+
+What makes Autoformer unique? It doesn't treat trend and seasonal patterns as noise — it learns to **decompose** them, improving both performance and interpretability. Plus, it's super efficient for **long-term forecasting**, avoiding the bottlenecks of traditional Transformer-based approaches.
+
+It’s especially great for scenarios like energy consumption, traffic flow, economic indicators, weather patterns, and disease spread forecasting.
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:29]:
```suggestion
[Autoformer](https://huggingface.co/papers/2106.13008) is a model designed for long-term forecasting. It modifies the Transformer architecture with decomposition blocks that breaks down long-term trends, and it also adds an Auto-Correlation mechanism in place of self-attention to discover repeating patterns. The model combines these two designs to make better predictions.
```

File: docs/source/en/model_doc/autoformer.md [hunk 2]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+
+You can explore the model on the [Autoformer Hub Page](https://huggingface.co/elisim/autoformer-energy) or other community-hosted variants.
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:31]:
```suggestion
You can find all the original Autoformer checkpoints on the [Hub](https://huggingface.co/models?search=autoformer).
```

File: docs/source/en/model_doc/autoformer.md [hunk 3]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+
+> [!TIP]
+> Click on the Autoformer model cards in the right sidebar to explore more examples and use cases like weather or energy time series forecasting.
+
+The example below demonstrates how to generate text with [`Pipeline`], [`AutoModel`]
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:36]:
```suggestion
The example below demonstrates how to make a prediction with [`AutoformerForPrediction`].
```

File: docs/source/en/model_doc/autoformer.md [hunk 4]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:39]: Lets remove this since the `Pipeline` doesn't support forecasting

File: docs/source/en/model_doc/autoformer.md [hunk 5]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+
+```py
+from transformers import pipeline
+
+forecast = pipeline(task="time-series-forecasting", model="elisim/autoformer-energy")
+output = forecast("energy.csv")  # this assumes a supported dataset format
+```
+</hfoption> 
+
+<hfoption id="AutoModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:49]:
```suggestion
<hfoption id="AutoformerForPrediction">
```

File: docs/source/en/model_doc/autoformer.md [hunk 6]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+# Wrap the tensor in a dictionary, assuming input is named "past_values" or similar
+output = model(past_values=dummy_input)  # Replace with correct input name if different
 
-## Overview
+print(output)
+```
+</hfoption> 
 
-The Autoformer model was proposed in [Autoformer: Decomposition Transformers with Auto-Correlation for Long-Term Series Forecasting](https://arxiv.org/abs/2106.13008) by Haixu Wu, Jiehui Xu, Jianmin Wang, Mingsheng Long.
+<hfoption id="transformers-cli">
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:70]: Also remove this since its unsupported

File: docs/source/en/model_doc/autoformer.md [hunk 7]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
 
-This model augments the Transformer as a deep decomposition architecture, which can progressively decompose the trend and seasonal components during the forecasting process.
+```bash
+transformers-cli env  # transformers-cli doesn't currently support time series inputs
+```
+</hfoption> 
+</hfoptions>
 
-The abstract from the paper is the following:
+Quantization
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:78]: Lets also remove the Quantization content and Attention Mask Visualization stuff since none of that is supported either

File: docs/source/en/model_doc/autoformer.md [hunk 8]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+Attention Mask Visualization
+Autoformer currently does not use standard attention mechanisms, so it isn't compatible with AttentionMaskVisualizer. Instead, it uses an Auto-Correlation mechanism, which doesn't operate on per-token attention masks.
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+## Notes
 
-- Check out the Autoformer blog-post in HuggingFace blog: [Yes, Transformers are Effective for Time Series Forecasting (+ Autoformer)](https://huggingface.co/blog/autoformer)
+- Autoformer was proposed in the paper [Autoformer: Decomposition Transformers with Auto-Correlation for Long-Term Series Forecasting](https://arxiv.org/abs/2106.13008) by Haixu Wu et al.
+- The model was contributed by [@elisim](https://huggingface.co/elisim) and [@kashif](https://huggingface.co/kashif).
+- The official implementation is available on [GitHub](https://github.com/thuml/Autoformer).
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:96]: Lets also remove these `## Notes` since they aren't directly related to usage.

```suggestion
```

File: docs/source/en/model_doc/autoformer.md [hunk 9]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+- Check out Hugging Face’s blog post on this model: [Yes, Transformers are Effective for Time Series Forecasting (+ Autoformer)](https://huggingface.co/blog/autoformer).
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:97]:
```suggestion
- Read the [Yes, Transformers are Effective for Time Series Forecasting (+ Autoformer)](https://huggingface.co/blog/autoformer) blog post for more details about using Autoformer to make predictions.
```

File: docs/source/en/model_doc/autoformer.md [hunk 10]
```diff
@@ -46,9 +103,9 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 ## AutoformerModel
 
 [[autodoc]] AutoformerModel
-    - forward
+- forward
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:106]: These don't need to be changed

File: docs/source/en/model_doc/autoformer.md [hunk 11]
```diff
@@ -14,30 +14,87 @@ rendered properly in your Markdown viewer.
...
+
+forecast = pipeline(task="time-series-forecasting", model="elisim/autoformer-energy")
+output = forecast("energy.csv")  # this assumes a supported dataset format
+```
+</hfoption> 
+
+<hfoption id="AutoModel">
+
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/autoformer.md:52]: Lets use this example here:

```py
import torch
from huggingface_hub import hf_hub_download
from transformers import AutoformerForPrediction, AutoformerConfig

file = hf_hub_download(
    repo_id="hf-internal-testing/tourism-monthly-batch", filename="train-batch.pt", repo_type="dataset"
)
batch = torch.load(file)

model = AutoformerForPrediction.from_pretrained("huggingface/autoformer-tourism-monthly")

outputs = model.generate(
    past_values=batch["past_values"],
    past_time_features=batch["past_time_features"],
    past_observed_mask=batch["past_observed_mask"],
    static_categorical_features=batch["static_categorical_features"],
    future_time_features=batch["future_time_features"],
)

mean_prediction = outputs.sequences.mean(dim=1)
print(f"Mean prediction: {mean_prediction}")
```

**Review by stevhliu — COMMENTED** [review]
Nice work, excited to have our first updated time-series model card! 🎉

File: docs/source/en/model_doc/autoformer.md (hunk 1 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md]: Done

File: docs/source/en/model_doc/autoformer.md (hunk 2 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md]: Done

File: docs/source/en/model_doc/autoformer.md (hunk 4 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md]: Done

File: docs/source/en/model_doc/autoformer.md (hunk 5 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md]: Done

File: docs/source/en/model_doc/autoformer.md (hunk 6 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md]: Done

File: docs/source/en/model_doc/autoformer.md (hunk 7 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md:65]: Done

File: docs/source/en/model_doc/autoformer.md (hunk 10 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md]: Reverted. Done

File: docs/source/en/model_doc/autoformer.md (hunk 11 above)
Comment (AnupKulkarniVK) [reply docs/source/en/model_doc/autoformer.md:39]: Updated the example. Done

**Comment by AnupKulkarniVK** [comment]
@stevhliu take care of the review comments. Please take a look. Thanks

File: docs/source/en/model_doc/autoformer.md (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/model_doc/autoformer.md]: Not resolved yet! I think its better to leave it as how to make a prediction since the docs don't explicitly show how to use it for things like weather and energy

File: docs/source/en/model_doc/autoformer.md (hunk 7 above)
Comment (stevhliu) [reply docs/source/en/model_doc/autoformer.md:65]: This hasn't been removed yet!

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more unresolved changes and then we can merge!
